pycytominer_transform/convert.py

```diff
@@ -2,349 +2,254 @@
-        records=records, source_datatype=source_datatype
-    )
-
-    # filter source filepaths to inferred or source datatype
-    return filter_source_filepaths(records=records, source_datatype=source_datatype)
+    return source
 
 
 @task
-def read_file(record: Dict[str, Any]) -> Dict[str, Any]:
+def prepend_column_name(
+    source: Dict[str, Any],
+    source_group_name: str,
+    identifying_columns: Union[List[str], Tuple[str, ...]],
+    metadata: Union[List[str], Tuple[str, ...]],
+    targets: List[str],
+) -> Dict[str, Any]:
     """
-    Read csv file from record.
+    Rename columns using the source group name, avoiding identifying columns.
 
     Args:
-      record: Dict[str, Any]:
-        Data containing filepath to csv file
+        source: Dict[str, Any]:
+            Individual data source source which includes meta about source
+            as well as Arrow table with data.
+        source_group_name: str:
+            Name of data source source group (for common compartments, etc).
+        identifying_columns: Union[List[str], Tuple[str, ...]]:
+            Column names which are used as ID's and as a result need to be
+            ignored with regards to renaming.
+        metadata: Union[List[str], Tuple[str, ...]]:
+            List of source data names which are used as metadata
+        targets: List[str]:
+            List of source data names which are used as compartments
 
     Returns:
-      record: Dict[str, Any]
-        Updated record (Dict[str, Any]) with CSV data in-memory
+        Dict[str, Any]
+            Updated source which includes the updated table column names
     """
 
-    if AnyPath(record["source_path"]).suffix == ".csv":  # pylint: disable=no-member
-        # define invalid row handler for rows which may be
-        # somehow erroneous. See below for more details:
-        # https://arrow.apache.org/docs/python/generated/pyarrow.csv.ParseOptions.html#pyarrow-csv-parseoptions
-        def skip_erroneous_colcount(row):
-            return "skip" if row.actual_columns != row.expected_columns else "error"
-
-        # setup parse options
-        parse_options = csv.ParseOptions(invalid_row_handler=skip_erroneous_colcount)
-
-        # read csv using pyarrow lib and attach table data to record
-        record["table"] = csv.read_csv(
-            input_file=record["source_path"],
-            parse_options=parse_options,
+    # stem of source group name
+    # for example:
+    #   targets: ['cytoplasm']
+    #   source_group_name: 'Per_Cytoplasm.sqlite'
+    #   source_group_name_stem: 'Cytoplasm'
+    source_group_name_stem = targets[
+        # return first result from generator below as index to targets
+        next(
+            i
+            for i, val in enumerate(targets)
+            # compare if value from targets in source_group_name stem
+            if val.lower() in str(pathlib.Path(source_group_name).stem).lower()
         )
+        # capitalize the result
+    ].capitalize()
```

Comment:
I assume this code is to detect a plausible prefix for the columns based on the source filename, right? Would it perhaps make more sense to just have the user supply this prefix directly, rather than having it inferred from a combination of `targets ` and `source_group_name`?

Reply:
This could be added into a configuration preset or parameter of some kind. I sought to avoid doing so because of the existing number of parameters and the patterns I've seen thus far in prepended column naming.

Given the example:

```markdown
- targets: ['cytoplasm']
- source_group_name: 'Per_Cytoplasm.sqlite'
- source_group_name_stem: 'Cytoplasm'
```

`source_group_name` could also be `'Cytoplasm.csv'` or `'cytoplasm.csv'` so it felt correct to make this work easier on the scientist or developer making use of the package through inference. Generally, there might be a need to create options for output or input data schema specification (including things like column names and datatypes), but I don't yet know how this would operate or whether it'd be helpful (it could be too constraining or complex).

Reply:
Added #27 on this topic. Resolving this comment for now.